frontend/lib/src/components/shared/StreamlitMarkdown/styled-components.ts

```diff
@@ -134,65 +134,63 @@ export const StyledStreamlitMarkdown =
     }
   )
 
-export const StyledLinkIconContainer = styled.div(() => ({
-  position: "relative",
-  left: "calc(-2.5rem - 0.5rem)",
-  width: "calc(100% + 2.5rem + 0.5rem)",
-  display: "flex",
-  alignItems: "center",
-  overflow: "visible",
+export const StyledHeadingWithActionElements = styled.div(({ theme }) => ({
+  "h1, h2, h3, h4, h5, h6, span": {
+    scrollMarginTop: theme.spacing.threeXL,
+  },
+  ...sharedMarkdownStyle(theme),
+
+  // break-word & pretty makes most headings break in a nicer way than break-all while still
+  // preventing overflowing of the container to the side. Long headings without whitespaces or hyphens might still look weird
+  wordBreak: "break-word",
+  textWrap: "pretty",
+
+  // show link-icon when hovering somewhere over the heading
+  // we use opacity instead of visibility because the react-testing-library seems
+  // to have issues with queryByRole('link') otherwise, even when triggering hover-events
+  "& .stHeaderActionElements > a": {
```

Comment:
nit: I think there is a way to actually directly link the `StyledHeadingActionElements` here instead of using a class name, e.g.:

https://github.com/streamlit/streamlit/blob/66c7aa235c311a8c253de3d5981c29afb934ae17/frontend/lib/src/components/elements/AlertElement/styled-components.tsx#L29

Reply:
Ah neat, thanks!